mayastor/src/bdev/nvmx/controller.rs

```diff
@@ -821,22 +822,36 @@ extern "C" fn aer_cb(ctx: *mut c_void, cpl: *const spdk_nvme_cpl) {
     }
 }
 
-/// return number of completions processed (maybe 0) or the negated on error,
-/// which is one of:
-///
-/// ENXIO: the qpair is not conected  or when the controller is
-/// marked as failed.
-///
-/// EGAIN: returned whenever the controller is being reset.
+/// Poll to process qpair completions on admin queue
+/// Returns: 0 (SPDK_POLLER_IDLE) or 1 (SPDK_POLLER_BUSY)
 pub extern "C" fn nvme_poll_adminq(ctx: *mut c_void) -> i32 {
     let mut context = NonNull::<TimeoutConfig>::new(ctx.cast())
         .expect("ctx pointer may never be null");
     let context = unsafe { context.as_mut() };
 
+    // returns number of completions processed (maybe 0) or the negated error,
+    // which is one of:
+    //
+    // ENXIO: the qpair is not connected or when the controller is
+    // marked as failed.
+    //
+    // EAGAIN: returned whenever the controller is being reset.
     let result = context.process_adminq();
 
     if result < 0 {
-        //error!("{}: {}", context.name, Errno::from_i32(result.abs()));
+        error!(
+            "process adminq: {}: {}",
+            context.name,
+            Errno::from_i32(result.abs())
+        );
+        if context.start_device_destroy() {
```

Comment:
Can we add a new message to denote the scheduling of device removal, so that it's clear that the operation has been started ? Straight after the 'if ...' ?
We could also display here the error that caused the failure and it will be printed only once without flooding the console (not sure if we care about all the subsequent errors since the device was identified as faulted and removal has been scheduled anyways).

Reply:
Good idea, will do.